src/test/scala/it/pagopa/interop/backendforfrontend/AuthorizationApiServiceSpec.scala

```diff
@@ -237,6 +383,187 @@ class AuthorizationApiServiceSpec extends AnyWordSpecLike with SpecHelper with S
       }
     }
 
+    "succeed if the tenant needs to be upserted and its origin is not IPA but is in allowlist" in {
+
+      val uid: String        = UUID.randomUUID().toString
+      val selfcareId: String = allowList.head
+      val tenantId: UUID     = UUID.randomUUID()
+
+      val jwtClaimsSet: JWTClaimsSet = Map[String, Object](
+        "organization" -> Map(
+          "id"         -> selfcareId,
+          "fiscalCode" -> "fiscalCode",
+          "roles"      -> List(Map("role" -> "admin"), Map("role" -> "anotherRole"))
+        ),
+        "uid"          -> uid
+      ).asClaimSet
+
+      (mockJwtReader
+        .getClaims(_: String))
+        .expects(*)
+        .once()
+        .returns(Success(jwtClaimsSet))
+
+      (mockInteropTokenGenerator
+        .generateInternalToken(_: String, _: List[String], _: String, _: Long))
+        .expects(*, *, *, *)
+        .once()
+        .returns(Future.successful(internalToken))
+
+      (mockTenantManagement
+        .getBySelfcareId(_: String)(_: Seq[(String, String)]))
+        .expects(selfcareId, *)
+        .once()
+        .returns(Future.failed(ApiError[Problem](404, "oh no!", None, new Exception, Map.empty)))
+
+      (mockPartyProcess
+        .getInstitution(_: String)(_: Seq[(String, String)], _: ExecutionContext))
+        .expects(selfcareId, *, *)
+        .once()
+        .returns(
+          Future.successful(
+            Institution(
+              id = UUID.fromString(selfcareId),
+              externalId = "whatever",
+              originId = "non-IPACode",
+              description = "foo",
+              digitalAddress = "of a digital home?",
+              address = "of an actual home?",
+              zipCode = "winzip",
+              taxCode = "not mine please",
+              origin = "test-code",
```

Comment:
just for consistency with other tests and for improve readability
(below values should be changed as well)
```suggestion
              origin = "non-IPA",
```